internal/marketplaces/bundles/sources/source.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 Stacklok, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package sources contains logic for loading a bundle from a source of bundles
+package sources
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/stacklok/minder/pkg/mindpak"
+	"github.com/stacklok/minder/pkg/mindpak/reader"
+)
+
+// BundleSource contains methods for retrieving bundles. Implementations may
+// load the Bundle from disk, an OCI registry, or any other place where bundles
+// may be present. Bundles are returned as instances of the BundleReader
+// interface.
+type BundleSource interface {
```

Comment:
Can we have the BundleSource interfaces and implementations in the pkg/mindpak package? I would love to have them public to work on external tools to manage them.